In [40]:
from IPython.display import HTML

def ro(s, d=2):
    return str(round(s, d))

HTML('''<script>
code_show=true; 
function code_toggle() {
 if (code_show){
 $('div.input').hide();
 } else {
 $('div.input').show();
 }
 code_show = !code_show
} 
$( document ).ready(code_toggle);
</script>
<style>
.output_subarea pre {
    background-color: red;
    word-break: normal !important;
    word-wrap: normal !important;
    text-align: left;
    white-space: break-spaces;
}
</style>
<form action="javascript:code_toggle()"><input type="submit" value="Click here to toggle on/off the raw code."></form>''')



# Carbon Dioxide Deposit Size 


## Introduction

__Mol__ 
is defined as the amount of a chemical substance that contains as many representative particles (atoms, molecules, ions, electrons, or photons) as there are atoms in 12 grams of carbon-12.

1 mole has `12.011 g C`, and 1 gram of Carbon has 0.083257 mole.
1 mole C/liter = 12.011 × 10-3 Gt C/km3

__Mass of the Earth's atmosphere__
is ([Mass of the Earth Estimates](https://arxiv.org/ftp/arxiv/papers/1810/1810.08658.pdf)):

**5.137 × 10<sup>15</sup> tonnes** = 5.137 × 10<sup>18</sup>kg  
Bottom estimate: 5.137972E+18kg  
Upper  estimate: 5.150502E+18kg
  



__Carbon atomic weight__ ([Carbon](https://en.wikipedia.org/wiki/Carbon) / 
[Isotopes of carbon](https://en.wikipedia.org/wiki/Isotopes_of_carbon))  

Standard atomic weight Ar, std(C)	[12.0096, 12.0116] conventional: `12.011`
- Using the abundance of Carbon isotopes, we calculate the average atomic weight of Carbon.

| name | fraction | weight | % | AVG |
| ---- |-------|--------------|------:|-------------:|
|  12C | 0.989 | 12.00000000  | 98.9% | 1186.80000000 |
|  13C | 0.011 | 13.00335484  | 1.1%  |   14.30369032 |
|    - |     - | Total:       | 100%  |   12.01103690 |



__Oxygen atomic weight__ (
[Oxygen](https://en.wikipedia.org/wiki/Oxygen) / 
[Isotopes_of_oxygen](https://en.wikipedia.org/wiki/Isotopes_of_oxygen))  

Standard atomic weight Ar, std(O)	[15.99903, 15.99977] conventional: `15.999`

- Using the abundance of Oxigen isotopes, we calculate the average atomic weight of Oxigen.

| name | fraction | weight | % | AVG |
| ---- | -------- | ------------ | -----:|-------------:|
| 16O  | 0.9976 | 15.99491462 |	99.76% |	1595.652682000 |
| 17O  | 0.0004	| 16.99913176 |	 0.04% |	   0.679965270 |
| 18O  | 0.0020	| 17.99915961 |   0.2% |   3.599831923 |
|    - |      - | Total:       | 100%   |   15.999324800 |


In [41]:
class bcolors:
    HEADER = '\033[95m'
    OKBLUE = '\033[94m'
    OKGREEN = '\033[92m'
    WARNING = '\033[93m'
    FAIL = '\033[91m'
    ENDC = '\033[0m'
    BOLD = '\033[1m'
    UNDERLINE = '\033[4m'

mAtm_ton=5.137 * 10**15
Cm=12.0107
Om=15.999
co2m=Cm+(Om*2)
C_in_co2_pc=round((Cm/co2m)*100,4)
print ("1 mole of CO₂ therefore has a molecular mass of " + ro(co2m,4) + "." )
print ("(C) Carbon content in (CO₂) Carbon-Dioxide is " + ro(C_in_co2_pc) + "%.")

1 mole of CO₂ therefore has a molecular mass of 44.0087.
(C) Carbon content in (CO₂) Carbon-Dioxide is 27.29%.


## Current CO₂ in atmosphere by NASA/NOAA  
([CO2 mesurement NASA](https://climate.nasa.gov/vital-signs/carbon-dioxide/), 
[CO2 mesurement NOAA - file](ftp://aftp.cmdl.noaa.gov/products/trends/co2/co2_mm_mlo.txt))  

In [42]:
import numpy as np
import pandas as pd

data = [
    [2015, 5, 403.96, "-"],
    [2016, 5, 407.72, 3.76],
    [2017, 5, 409.69, 1.97],
    [2018, 5, 411.24, 1.55],
    [2019, 5, 414.66, 3.42] 
]
pd.DataFrame(data, columns=["Year", "Month", "CO2ppmv", "ppmv inc."])

Year  Month  CO2ppmv ppmv inc.
0  2015      5   403.96         -
1  2016      5   407.72      3.76
2  2017      5   409.69      1.97
3  2018      5   411.24      1.55
4  2019      5   414.66      3.42

In [43]:
CO2avg_inc=round((3.76 + 1.97 + 1.55 + 3.42)/4,2)
CO2ppmv=round(414.66+CO2avg_inc,2)
CO2per=CO2ppmv/(100*100)
CO2perYear=2.675

print ("We can estimate that average growth for the last 5 years is {} ppmv per year. ".format(CO2avg_inc) + 
       "\nEstimated CO₂ for May 2020 will be {}.".format(CO2ppmv) )
print ("Taking the highest measured concentration of CO₂ as " + str(CO2ppmv) + " ppmv, "
        + "equivalent to percentage content of " + str(CO2per) + "%.", sep="", end="", flush=True)

We can estimate that average growth for the last 5 years is 2.67 ppmv per year. 
Estimated CO₂ for May 2020 will be 417.33.
Taking the highest measured concentration of CO₂ as 417.33 ppmv, equivalent to percentage content of 0.041733%.

## Size Calculation

Now, we can calculate the following:  
W = 44.0087 g/mol (CO2 molecular mass)  
A = 28.971 g/mole ([average molar mass of dry air](https://en.wikipedia.org/wiki/Molar_mass#Average_molar_mass_of_mixtures))

In [44]:
avgDryAir=28.971
co2m_pc=CO2per*(co2m/avgDryAir)
co2inA=round((mAtm_ton*(co2m_pc/100)) / 1000**3, 4)
print ("CO2% = {CO2per}% (CO2 atmosphere volume)".format(CO2per=CO2per))  
print (bcolors.BOLD + "CO2 m% = CO2% * (W/A) = {co2m_pc}".format(co2m_pc=co2m_pc))

print (bcolors.ENDC + "Multiplying Earth’s atmosphere {} tonnes * {} m% CO2,".format(mAtm_ton,co2m_pc) + 
       "\nas a result the "+ bcolors.BOLD +" total weight of CO2 in the atmosphere is " + str(co2inA) + " Gt CO2.")

CO2% = 0.041733% (CO2 atmosphere volume)
CO2 m% = CO2% * (W/A) = 0.06339494933209072
Multiplying Earth’s atmosphere 5137000000000000.0 tonnes * 0.06339494933209072 m% CO2,
as a result the  total weight of CO2 in the atmosphere is 3256.5985 Gt CO2.


In [45]:
C_Gt=(co2inA/CO2ppmv)*(C_in_co2_pc/100)
print ("Every 1 ppmv (part per million of volume) of the atmosphere weighs approximately " + ro(C_Gt,4) + \
       " Gt of Carbon.")

Every 1 ppmv (part per million of volume) of the atmosphere weighs approximately 2.1297 Gt of Carbon.


Scientists have warned that reaching `450 ppmv would mean a 50% chance of heating planet 2C` - which many call the `point of no return`.

In [46]:
CO2_stableLevel=350
CO2_endGame=450
optimistic_time_left=(CO2_endGame-CO2ppmv)/CO2perYear
print ("Having in mind above estimate of CO2 levels {CO2ppmv} ppmv we will reach in ".format(CO2ppmv=CO2ppmv) +
"May 2020 Earth's coupling with factual detrimental events as Australian wildfires, storms, floods, " +
"hurricanes, droughts and especially thawing permafrost in Russia, we can calculate that we approximately " + 
"have "+ bcolors.BOLD +"less than " +  ro(optimistic_time_left) + 
" years within which we need to put our best effort to prevent human extinction." + bcolors.ENDC)


Having in mind above estimate of CO2 levels 417.33 ppmv we will reach in May 2020 Earth's coupling with factual detrimental events as Australian wildfires, storms, floods, hurricanes, droughts and especially thawing permafrost in Russia, we can calculate that we approximately have less than 12.21 years within which we need to put our best effort to prevent human extinction.


The pre-industrial CO2 concentration in the atmosphere, for instance in the year 1800 was around 280 ppmv bu 350 ppmv has been generally accepted as the maximum stable CO2 level. So, let’s now consider how much CO2 or Carbon we have added since the Industrial Revolution and how much carbon we need to sequester, to be safe.

In [47]:
CO2_stableLevel=350
CO2_preIndustrial=280
CO2_humans_added=CO2ppmv-CO2_preIndustrial
C_human_added_mass=round(CO2_humans_added*C_Gt,2)
CO2_humans_safe=CO2ppmv-CO2_stableLevel
C_human_safe_mass=round(CO2_humans_safe*C_Gt, 2)
print ("Human influence has added " + ro(CO2_humans_added) + " ppmv which equates to " + \
    str(C_human_added_mass) + " Gigatonnes of Carbon.")
print ("To go back to the safe level of "+str(CO2_stableLevel) + \
    " ppmv, we would need to sequester " + bcolors.OKBLUE + bcolors.BOLD + str(C_human_safe_mass) + " GtC")   

Human influence has added 137.33 ppmv which equates to 292.47 Gigatonnes of Carbon.
To go back to the safe level of 350 ppmv, we would need to sequester 143.39 GtC


If we take the current [world population](https://en.wikipedia.org/wiki/World_population)  of 7.7 billion people as of April 2019, and we give each person to take care of hers equal portion of Carbon, each person would need to sequester around:    

In [48]:
giga=1000**3
ton2kg=1000
world_population = 7.7 * giga
C_per_human=round((C_human_safe_mass*giga*ton2kg)/world_population, 0)
print ((str(int(C_per_human)) + "kg of Carbon!"))

brick_vol_m3=0.00078657907
brick_weight=round(brick_vol_m3 * 2266, 3)
# print (brick_weight)
bricks_per_human=round(C_per_human/brick_weight, 0)
brick_per_m3=1/brick_vol_m3
m3_per_human=bricks_per_human/brick_per_m3

18622kg of Carbon!


Common brick of size 8 by 3 by 2 inches, has a volume of 48 cubic Inch or 0.00078657907 cubic meters. If bricks were made out of **graphite** (pure Carbon) it would have a density 2266 kg/m³, then single brick would weigh 1.782 kilograms.

In [49]:
print ("Graphite brick would weight ", brick_weight, "kg, and then each human would need to sequester " + \
    str(int(bricks_per_human)) + " bricks. ")
print ("As in one cubic meter cube, we could fit " + str(int(brick_per_m3)) + \
       " bricks each human would need to sequester " + \
 str(round(m3_per_human,1)) + " cubic meters.")

Graphite brick would weight  1.782 kg, and then each human would need to sequester 10450 bricks. 
As in one cubic meter cube, we could fit 1271 bricks each human would need to sequester 8.2 cubic meters.


On the other hand, if we would keep the sequestered Carbon store in blocks of wood as Pinewood for instance, which is a 'medium' density softwood (400-420 kg/m³) with Carbon content inside of the wood ranging between 45%-50%.

In [50]:
pine_wood_density=400
pine_carbon_pc=45
pine_carbon_per_m3=pine_wood_density*(pine_carbon_pc/100)
total_pine_m3=C_per_human/pine_carbon_per_m3
standard_room_volume=4*4*2.5
rooms_total=total_pine_m3/standard_room_volume
print ("We would need " + ro(total_pine_m3,1), "cubic meters. Or in other words, if we take the standard room " 
       + "(4m long x 4m wide x 2.5m high), we would need to fill "
       + ro(rooms_total,1) + " of such rooms with Pine lumber.")

We would need 103.5 cubic meters. Or in other words, if we take the standard room (4m long x 4m wide x 2.5m high), we would need to fill 2.6 of such rooms with Pine lumber.


[The Great Wall of China](https://en.wikipedia.org/wiki/Great_Wall_of_China) extends about 21,196 km with the wall around 7.8m high and 5m wide.

In [51]:
great_w_height=7.8
great_w_width=5
great_w_length=21196*1000
great_wall_vol=great_w_height*great_w_width*great_w_length
print ("It has a total volume of " + str(int(great_wall_vol)), "cubic meters. ")
bricks_vol_humanity=world_population*m3_per_human
print (bcolors.OKBLUE + "As humanity needs to sequester ", str(int(round(bricks_vol_humanity,0))) + \
    " m3 Carbon - equivalent would be " + \
    bcolors.OKBLUE + bcolors.BOLD + \
    "a volume of " + str(round(bricks_vol_humanity/great_wall_vol,1)), "Great China Walls in bricks." + bcolors.ENDC) 
earth_circumference_m=40075*1000
volume_earth_wall=earth_circumference_m*10*5;
print ("As Earth's circumference around the equator is 40075km, to turn enough atmospheric Carbon to" + \
    " get to stable 350 ppmv in graphite bricks - we could build a wall 3 story high (10 meters)" + \
    " and 5 meters wide, that would " + bcolors.OKBLUE + bcolors.BOLD + "encircle the Earth around Equator "  + \
       str(int(round(bricks_vol_humanity/volume_earth_wall,0))) + " times."
      )


It has a total volume of 826644000 cubic meters. 
As humanity needs to sequester  63292084868 m3 Carbon - equivalent would be a volume of 76.6 Great China Walls in bricks.
As Earth's circumference around the equator is 40075km, to turn enough atmospheric Carbon to get to stable 350 ppmv in graphite bricks - we could build a wall 3 story high (10 meters) and 5 meters wide, that would encircle the Earth around Equator 32 times.


### Is there a way to do such a massive endeavour?
Yes, there is a way, by simply helping it to build itself.

*<sub>Our next articles provide more details.</sub>